src/libs/actions/IOU.ts

```diff
@@ -7008,6 +7009,24 @@ function isLastApprover(approvalChain: string[]): boolean {
     return approvalChain[approvalChain.length - 1] === currentUserEmail;
 }
 
+function getNextApproverAccountID(report: OnyxEntry<OnyxTypes.Report>) {
+    const ownerAccountID = report?.ownerAccountID ?? -1;
+    const policy = PolicyUtils.getPolicy(report?.policyID);
+    const approvalChain = ReportUtils.getApprovalChain(policy, ownerAccountID, report?.total ?? 0);
+    const submitToAccountID = PolicyUtils.getSubmitToAccountID(policy, ownerAccountID);
+
+    if (approvalChain.length === 0) {
+        return submitToAccountID;
```

Comment:
do we need any additional checks for submitAccountID?

Reply:
FYI this logic is taken from how we get the next approver display name to build next step here. As what we are aiming to achieve is to set the managerID to the next approver and as `getSubmitToAccountID` always returns account  `number` we don't need any check
https://github.com/Expensify/App/blob/00bc167940607f1418109f00e2b4e2d7d4ae858b/src/libs/actions/IOU.ts#L7030
https://github.com/Expensify/App/blob/00bc167940607f1418109f00e2b4e2d7d4ae858b/src/libs/NextStepUtils.ts#L96
https://github.com/Expensify/App/blob/00bc167940607f1418109f00e2b4e2d7d4ae858b/src/libs/NextStepUtils.ts#L63-L74

Reply:
If we got the logic from there can we refactor this so we don't need to maintain two very similar utils?

Reply:
Maybe the `getNextApproverDisplayName()` can get the `nextApproverAccountID` and use it to get the `displayName`?